Src/Witsml/Metrics/WitsmlMetrics.cs

```diff
@@ -0,0 +1,64 @@
+using System;
+using System.Diagnostics;
+using System.Diagnostics.Metrics;
+using System.Reflection;
+using System.Threading.Tasks;
+
+using Witsml.ServiceReference;
+
+namespace Witsml.Metrics;
+
+internal sealed class WitsmlMetrics
+{
+    #region Singleton
+    // We still want to collect unified metrics in case there are multiple WitsmlClient instances used
+
+    private static readonly Lazy<WitsmlMetrics> LazyInstance =
+        new(() => new WitsmlMetrics());
+
+    public static WitsmlMetrics Instance { get { return LazyInstance.Value; } }
+
+    private WitsmlMetrics() { }
+
+    #endregion
+
+    private static readonly AssemblyName AssemblyName =
+        typeof(WitsmlMetrics).Assembly.GetName();
+
+    internal static readonly string MeterName = AssemblyName.Name;
+
+    private static readonly Meter MeterInstance = new(MeterName, AssemblyName.Version!.ToString());
+
+    private readonly Histogram<long> _requestDuration = MeterInstance.CreateHistogram<long>(
+        "witsml.requests.duration",
+        unit: "s",
+        description: "Time spent during requests to a Witsml server");
+
+    private readonly UpDownCounter<int> _activeRequests =
+        MeterInstance.CreateUpDownCounter<int>(
+            "witsml.requests.active",
+            description: "Number of active requests");
+
+    internal async Task<TResponseType> MeasureQuery<TResponseType>(Uri serverUri, WitsmlMethod method, string witsmlType, Task<TResponseType> wmlsTask)
+        where TResponseType : IWitsmlResponse
+    {
+        var tagList = new TagList
+        {
+            { "host", serverUri.Host },
+            { "method", Enum.GetName(method) },
+            { "objectType", witsmlType },
+        };
+
+        _activeRequests.Add(1, tagList);
+        var timer = Stopwatch.StartNew();
+        TResponseType response = await wmlsTask;
```

Comment:
Just to verify: Are there any edge cases where wmlsTask would completely fail instead of returning a proper response? If it's possible, the _activeRequests wouldn't get reset.

I tested with malformatted requests and always got a response, but we might get problems if the server is unreachable. Sounds like a good idea to handle exceptions anyways.